src/report/report.py

```diff
@@ -350,6 +350,9 @@ def failure_matches_rule(
             if default_rule not in matching_rules:
                 matching_rules.append(default_rule)
 
+        # Sort matching_rules by the identical rules to failure.step
+        if matching_rules:
+            matching_rules = sorted(matching_rules, key=lambda x: x.step.__eq__(failure.step), reverse=True)
```

Comment:
@oharan2 just a quick edge case that is possible. Do you know if you had two rules with the exact step name, but different `failure_type` values, would it potentially cause some issues with that configuration?

Reply:
It'll eliminate the one with the wrong type [HERE](https://github.com/RedHatQE/firewatch/blob/main/src/objects/failure_rule.py#L195).

Added the check in [THIS](https://github.com/RedHatQE/firewatch/pull/193/commits/dc56944b5e13ce3c21b2d7dbaca3a491ea2388fa) commit.

Reply:
I actually think [the earlier](https://github.com/RedHatQE/firewatch/pull/193/commits/f94f4d684e43f812b37a88932c211105c738b7c3) version of the test might be clearer to understand later, do you think it's better to revert the latest commit?